pilot/pkg/model/validation.go

```diff
@@ -1547,14 +1547,19 @@ func checkServiceRoleBinding(in *rbac.ServiceRoleBinding) error {
 			errs = appendErrors(errs, fmt.Errorf("do not use * for names or not_names (in rule %d)", i))
 		}
 	}
-	if in.RoleRef != nil && len(in.Actions) > 0 {
-		errs = appendErrors(errs, fmt.Errorf("only one of `roleRef` or `actions` can be specified"))
+	if in.RoleRef != nil && in.Role != "" {
+		errs = appendErrors(errs, fmt.Errorf("only one of `roleRef`, `role`, or `actions` can be specified"))
```

Comment:
You only checked roleRef and role but the error message includes action?

Reply:
Yes, it's intentional. Regardless of the combination, I think we should let the user know that only one of `roleRef`, `role`, or `actions` can be specified.